classes/ObjectModel.php

```diff
@@ -1876,15 +1876,28 @@ public function deleteImage($force_delete = false)
      *
      * @return bool
      */
-    public static function existsInDatabase($id_entity, $table)
+    public static function existsInDatabase($id_entity, $table = null)
     {
+        $primary = 'id_' . bqSQL($table);
+
+        if ($table === null) {
+            $object_def = static::$definition;
+
+            if (!array_key_exists('table', $object_def) || !array_key_exists('primary', $object_def)) {
+                return false;
+            }
+
+            $table = $object_def['table'];
+            $primary = $object_def['primary'];
+        }
```

Comment:
```suggestion
        if ($table !== null) {
            $primary = 'id_' . bqSQL($table);
        } else {
            $object_def = static::$definition;

            if (!isset($object_ref['table']) || !isset($object_ref['primary'])) {
                return false;
            }

            $table = $object_def['table'];
            $primary = $object_def['primary'];
        }
```

Do not declare something you will not use if $table is null.
Always prefer isset instead of array_key_exists if you don't care null value, it's 2.5 faster.

Reply:
Merci !